server/channels/api4/user_test.go

```diff
@@ -7917,3 +7919,559 @@ func TestLoginWithDesktopToken(t *testing.T) {
+    })
+
+    t.Run("Get default profile image as system admin", func(t *testing.T) {
+        user := th.CreateUser()
+
+        img, resp, err := th.SystemAdminClient.GetDefaultProfileImage(context.Background(), user.Id)
+        require.NoError(t, err)
+        require.NotNil(t, img)
+        require.Equal(t, http.StatusOK, resp.StatusCode)
+    })
+
+    t.Run("Get default profile image with invalid user ID", func(t *testing.T) {
+		_, resp, err := th.Client.GetDefaultProfileImage(context.Background(), "invalid_user_id")
+		require.Error(t, err)
+		CheckNotFoundStatus(t, resp)
+	})
+
+    t.Run("Consistent default image for the same user", func(t *testing.T) {
+		user := th.CreateUser()
+
+		// Login as the newly created user
+		th.Client.Login(context.Background(), user.Email, user.Password)
+
+		img1, resp, err := th.Client.GetDefaultProfileImage(context.Background(), user.Id)
+		require.NoError(t, err)
+		CheckOKStatus(t, resp)
+
+		img2, resp, err := th.Client.GetDefaultProfileImage(context.Background(), user.Id)
+		require.NoError(t, err)
+		CheckOKStatus(t, resp)
+
+		require.Equal(t, img1, img2, "Default profile images should be consistent for the same user")
+
+		// Logout after the test
+		th.Client.Logout(context.Background())
+	})
+
+}
+
+func TestGetUserThread(t *testing.T) {
+	th := Setup(t).InitBasic()
+	defer th.TearDown()
+
+	client := th.Client
+	user := th.BasicUser
+	team := th.BasicTeam
+
+	t.Run("get thread for user", func(t *testing.T) {
+		// Create a post
+		post, _, err := client.CreatePost(context.Background(), &model.Post{
+			ChannelId: th.BasicChannel.Id,
+			Message:   "Root message",
+		})
+		require.NoError(t, err)
+
+		// Create a reply to ensure thread membership
+		_, _, err = client.CreatePost(context.Background(), &model.Post{
+			ChannelId: th.BasicChannel.Id,
+			RootId:    post.Id,
+			Message:   "Reply",
+		})
+		require.NoError(t, err)
+
+		// Get the thread
+		thread, resp, err := client.GetUserThread(context.Background(), user.Id, team.Id, post.Id, false)
+		require.NoError(t, err)
+		CheckOKStatus(t, resp)
+		require.NotNil(t, thread)
+		require.Equal(t, post.Id, thread.PostId)
+		require.Equal(t, int64(1), thread.ReplyCount)
+	})
+
+	t.Run("get thread for user with extended info", func(t *testing.T) {
+		post, _, err := client.CreatePost(context.Background(), &model.Post{
+			ChannelId: th.BasicChannel.Id,
+			Message:   "Root message for extended info",
+		})
+		require.NoError(t, err)
+
+		// Create a reply to ensure thread membership
+		_, _, err = client.CreatePost(context.Background(), &model.Post{
+			ChannelId: th.BasicChannel.Id,
+			RootId:    post.Id,
+			Message:   "Reply for extended info",
+		})
+		require.NoError(t, err)
+
+		thread, resp, err := client.GetUserThread(context.Background(), user.Id, team.Id, post.Id, true)
+		require.NoError(t, err)
+		CheckOKStatus(t, resp)
+		require.NotNil(t, thread)
+		require.NotNil(t, thread.Participants)
+	})
```

Comment:
This test and the previous can be merged, also thread participants can be asserted as well.